skymp5-server/cpp/server_guest_lib/GetBaseActorValues.h

```diff
@@ -43,9 +46,51 @@ struct BaseActorValues
   }
 };
 
-inline BaseActorValues GetBaseActorValues(uint32_t baseId,
+namespace {
+void SetBaseActorValues(espm::LookupResult& result,
+                        espm::CompressedFieldsCache& compressedFieldsCache,
+                        BaseActorValues& baseActorValues)
+{
+  auto race = espm::Convert<espm::RACE>(result.rec);
+
+  auto raceData = race->GetData(compressedFieldsCache);
+
+  baseActorValues.health = raceData.startingHealth;
+  baseActorValues.magicka = raceData.startingMagicka;
+  baseActorValues.stamina = raceData.startingStamina;
+  baseActorValues.healRate = raceData.healRegen;
+  baseActorValues.magickaRate = raceData.magickaRegen;
+  baseActorValues.staminaRate = raceData.staminaRegen;
+}
+}
+
+inline BaseActorValues GetBaseActorValues(espm::Loader& espm, uint32_t baseId,
                                           uint32_t raceIdOverride)
 {
   BaseActorValues baseActorValues;
+  espm::CompressedFieldsCache compressedFieldsCache;
+
+  if (raceIdOverride) {
+    auto raceInfo = espm.GetBrowser().LookupById(raceIdOverride);
+
+    if (raceInfo.rec->GetType() == "RACE") {
+      SetBaseActorValues(raceInfo, compressedFieldsCache, baseActorValues);
+    } else {
+      throw std::runtime_error(
+        "Unable to read RACE. formId: " + std::to_string(baseId) +
+        ". raceId: " + std::to_string(raceIdOverride) + ".");
```

Comment:
Error message can be formatted like this:
`fmt::format("Unable to read RACE. formId: {}, raceId: {}", baseId, raceIdOverride)`

(`#include <fmt/format.h>`)

Another example in `skymp5-server/cpp/lib_espm/Combiner.cpp`